### PR TITLE
[WIP] add eval action

### PR DESCRIPTION
**Pull request by Mic92**
This is a POC to check wether it's feasible to replace ofborg with github actions runner.
Background is that we will loose the CI builder currently used for ofBorg.

### CODE REVIEW

**Comment by Yethal** [comment]
@Mic92 Is there a specific reason to use on push event and manually checking whether commit is part of PR instead of using on pull_request event directly?

**Comment by Mic92** [comment]
@Yethal yes, so that we have the runner limit of individual users instead of the NixOS org.

**Comment by JohnRTitor** [comment]
I'll suggest running merging this (keeping this alongside Ofborg) once it's stable enough. Then run it as a beta program for a few days, so we have sufficient confidence that it works seamlessly. 

And in seems like contributer forks branchs need to be updated to take advantage of this.

**Comment by infinisil** [comment]
New PR: https://github.com/NixOS/nixpkgs/pull/356023

For just evaluation (and those only taking 5 minutes on each arch) instead of also building, I don't think we need to do the running-on-forks dance.

File: .github/workflows/eval.yml [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    name: eval-check
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        system: [x86_64-linux, aarch64-linux, aarch64-darwin, x86_64-darwin]
+    steps:
+      - name: Set up Git
+        run: |
+          git config --global user.email "github-actions[bot]@users.noreply.github.com"
+          git config --global user.name "GitHub Actions"
```
Comment (TheGiraffe3) [inline .github/workflows/eval.yml:21]: One thing I've run into in the past is "global" can cause problems for other people.
Might it be worth switching to local (like this):
```suggestion
          git config --local user.name "GitHub Actions"
```
to not cause problems for other people?

Comment (FliegendeWurst) [reply .github/workflows/eval.yml:21]: It is "global" for the GH action run. So it doesn't really matter, I think

**Comment by infinisil** [comment]
Superseded by https://github.com/NixOS/nixpkgs/pull/356023

**Pull request closed without merging**